.github/workflows/build_and_test.yml

```diff
@@ -51,3 +51,48 @@ jobs:
         with:
           flags: unittests
           fail_ci_if_error: true # optional (default = false)
+
+  build_and_test:
```

Comment:
I think we can unify all jobs into one and just add a condition to `upload_code_coverage` step thatit should be executed only on Linux (but for now it should be still not active).